bundles/org.openhab.voice.whisperstt/README.md

```diff
@@ -0,0 +1,204 @@
+
+* https://huggingface.co/ggerganov/whisper.cpp
+* https://ggml.ggerganov.com
+
+You should place the downloaded .bin model in '\<openHAB userdata\>/whisper/' so the add-ons can find them.
+
+Remember to check that you have enough RAM to load the model, estimated RAM consumption can be checked on the huggingface link.
+
+## Using alternative whisper.cpp library
+
+It's possible to use your own build of the whisper.cpp shared library with this add-on.
+
+On `Linux/macOs` you need to place the `libwhisper.so/libwhisper.dydib` at `/usr/local/lib/`.
+
+On `Windows` the `whisper.dll` file can be placed at in any directory listed at the variable `$env:PATH`, for example `X:\\Windows\System32\`.
+
+In the [Whisper.cpp](https://github.com/ggerganov/whisper.cpp) README you can find information about the required flags to enable different acceleration methods on the cmake build and other relevant information.
+
+Note: You need to restart OpenHAB to reload the library.
+
+## Configuration
+
+Use your favorite configuration UI to edit the Whisper settings:
+
+### Speech to Text Configuration
+
+General options.
+
+* **Model Name** - Model name. The 'ggml-' prefix and '.bin' extension are optional here but required on the filename. (ex: tiny.en -> ggml-tiny.en.bin)
+* **Preload Model** - Keep whisper model loaded.
+* **Single Utterance Mode** - When enabled recognition stops listening after a single utterance.
+* **Max Transcription Seconds** - Max seconds for force trigger the transcription, without wait for detect silence.
+* **Initial Silence Seconds** - Max seconds without any voice activity to abort the transcription.
+* **Max Silence Seconds** - Max consecutive silence seconds to trigger the transcription.
+* **Remove Silence** - Remove start and end silence from the audio to transcribe.
+
+### Voice Activity Detection Configuration
+
+Configure VAD options.
+
+* **Audio Step** - Audio processing step in seconds for the voice activity detection.
+* **Voice Activity Detection Mode** - Selected VAD Mode.
+* **Voice Activity Detection Sensitivity** - Percentage in range 0-1 of voice activity in one second to consider it as voice.
+* **Voice Activity Detection Step** - VAD detector internal step in ms (only allows 10, 20 or 30). (Audio Step / Voice Activity Detection Step = number of vad executions per audio step).
+
+### Whisper Configuration
+
+Configure whisper options.
+
+* **Threads** - Number of threads used by whisper. (0 for default)
+* **Sampling Strategy** - Sampling strategy used.
+* **Beam Size** - Beam Size configuration for sampling strategy Bean Search.
+* **Greedy Best Of** - Best Of configuration for sampling strategy Greedy.
+* **Speed Up** - Speed up audio by x2. (Reduced accuracy)
+* **Audio Context** - Overwrite the audio context size. (0 for default)
+* **Temperature** - Temperature threshold.
+* **Initial Prompt** - Initial prompt for whisper.
+
+### Grammar Configuration
```

Comment:
Maybe earlier in the README, you could explain if configuring a grammar is mandatory or optional, and what is its purpose.
Isn't the grammar provided with the model ?
Maybe this should be clarified for users that are not very familiar with the Whisper stuff like you.

Reply:
I have wrote a section about the grammar feature. I'm not sure if it's a good explanation but there is not much written about it in the whisper.cpp/llama.cpp repos.

Reply:
Yes, it helps to understand.